### PR TITLE
feat(graphql): adding graphql

### PR DESCRIPTION
**Pull request by rhberro**
This pull request closes issue #62 

It adds a minimum support for GraphQL queries and mutations, also, added examples to the `README.md` file.

### CODE REVIEW

**Review by fsmaia — APPROVED** [review]

File: source/graphql.ts [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+function findQueryDefinitions(definition: any) {
+  return definition.name.value === 'Query';
+}
+
+function findMutationDefinitions(definition: any) {
+  return definition.name.value === 'Mutation';
+}
+
+export class GraphQLManager {
+  private definitions: any;
```
Comment (adrifmonte) [inline source/graphql.ts:46]: Rename to something like `typeDefs`, `gqlSchema`, or `apolloSchema` (avoiding `this.definitions.definitions`).

File: source/graphql.ts [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+  }
+
+  private getGraphQLMutations() {
+    return this.definitions.definitions.find(findMutationDefinitions);
+  }
+
+  private generateQueriesAndMutations(serverOptions: any) {
+    const serverOptionsEnhanced = {
+      ...serverOptions,
+      resolvers: {},
```
Comment (adrifmonte) [inline source/graphql.ts:71]: Could lift the `resolvers` to external API.

File: source/response/searchable.ts [hunk 3]
```diff
@@ -39,6 +39,7 @@ export default function createSearchableResponse(
   const { search, search: { parameter = 'search' } = {} } = method;
   const { query } = req;
 
+  // @ts-ignore
```
Comment (adrifmonte) [inline source/response/searchable.ts:42]: What's it ignoring?

File: source/interfaces/ServerOptions.ts [hunk 4]
```diff
@@ -5,10 +5,11 @@ import MethodOverride from './MethodOverride';
 
 export default interface ServerOptions {
   basePath?: string;
+  docsRoute?: string;
+  definitions?: TemplateStringsArray | string;
```
Comment (adrifmonte) [inline source/interfaces/ServerOptions.ts:9]: [style] Not in alphabetical order (also in other places).

**Review by adrifmonte — APPROVED** [review]

**Pull request merged**